deployment/deployment.go

```diff
@@ -0,0 +1,1224 @@
+	})
+}
+
+// DeleteDeploymentHealthCheckV2 - V2 version of space.DeleteAppHealthCheck
+// (original: "space/app.go")
+func DeleteDeploymentHealthCheckV2(db *sql.DB, params martini.Params, r render.Render) {
+	name := params["deployment"]
+	space := params["space"]
+
+	if _, err := db.Exec("UPDATE v2.deployments SET healthcheck=NULL WHERE name=$1 AND space=$2", name, space); err != nil {
+		utils.ReportError(err, r)
+		return
+	}
+
+	r.JSON(http.StatusOK, structs.Messagespec{Status: http.StatusOK, Message: "Deployment: " + name + "-" + space + " healthcheck removed"})
+}
+
+// DeleteDeploymentV2Handler - HTTP Handler for DeleteDeploymentV2
+func DeleteDeploymentV2Handler(db *sql.DB, params martini.Params, r render.Render) {
+	name := params["deployment"]
+	space := params["space"]
+
+	responseCode, err := DeleteDeploymentV2(db, name, space)
+	if err != nil {
+		if responseCode == 400 {
+			utils.ReportInvalidRequest(err.Error(), r)
+		} else {
+			utils.ReportError(err, r)
+		}
+		return
+	}
+
+	r.JSON(responseCode, structs.Messagespec{Status: responseCode, Message: name + "-" + space + " removed"})
+}
+
+// DeleteDeploymentV2 - V2 version of space.DeleteAppV2
+// (original: "space/app.go")
+func DeleteDeploymentV2(db *sql.DB, name string, space string) (int, error) {
+	exists, err := checkDeployment(db, name, space)
+	if err != nil {
+		return 500, err
+	}
+
+	if exists == false {
+		return 400, errors.New("Invalid app or space name")
+	}
+
+	rt, err := runtime.GetRuntimeFor(db, space)
+	if err != nil {
+		return 500, err
+	}
+
+	if err = rt.DeleteService(space, name); err != nil {
+		return 500, err
+	}
+
+	if err = rt.DeleteDeployment(space, name); err != nil {
+		return 500, err
+	}
+
+	rslist, err := rt.GetReplicas(space, name)
+	// in a previous iteration we ignored the error, i'll do so here
+	// but i need to put an if around it to prevent the nil rslist from
+	// bombing out everything
+	if err != nil {
+		fmt.Println(err)
+	}
+	if err == nil {
+		for _, rs := range rslist {
+			err = rt.DeleteReplica(space, name, rs)
+			if err != nil {
+				log.Println("Failed to remove replica set: ", err)
+			}
+		}
+	} else {
+		log.Println("Error getting replica sets:", err)
+	}
+
+	var podlist []string
+	podlist, err = rt.GetPods(space, name)
+	if err != nil {
+		fmt.Println(err)
+	}
+	// in a previous iteration we ignored the error, i'll do so here
+	// but i need to put an if around it to prevent the nil podlist from
+	// bombing out everything
+	if err == nil {
+		for _, pod := range podlist {
+			err = rt.DeletePod(space, pod)
+			if err != nil {
+				log.Println("Failed to remove pod: ", err)
+			}
+		}
+	} else {
+		log.Println("Error getting pods to remove:", err)
+	}
+	_, err = db.Exec("DELETE from appbindings where space=$1 and appname=$2", space, name)
+	if err != nil {
+		return 500, err
+	}
+
+	_, err = db.Exec("DELETE from v2.deployments where space=$1 and name=$2", space, name)
+	if err != nil {
+		return 500, err
```

Comment:
Can we use `http.Status*` codes instead of numbers?

Reply:
Fixed!